### DISCUSSION TITLE
TODOs before release

### DISCUSSION THREAD

**Discussion by ttlns** [Kaliiiiiiiiii-Vinyzu/CDP-Patches#3] [Ideas]
- [x] expose `base` 
- [x] allow dynamic scale factor (+ get on startup)
- [x] remove webdriver dependencies
- [x] move to `loop.run_in_executor` für asyncio for
  - https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/blob/7d99bcf1908fd9caaf218801460245bcb0af9661/cdp_patches/input/async_input.py#L72-L79
  - https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/blob/dd5555882387f689af29b4bc6634a0a8eb281825/cdp_patches/input/async_input.py#L87
- [x] add//support `timeout` kwarg
- [x] `asyncio.Lock` und `threading.Lock`
- [x] add Linux Support
- [x] Implement proper extensive Testing
- [x] document `base` & `scale_factor`
- [x] fix playwright wrong scale factor
  - DPI-awareness https://learn.microsoft.com/en-us/windows/win32/api/shellscalingapi/nf-shellscalingapi-getprocessdpiawareness
   - `ctypes.windll.user32.GetDpiForWindow(root.winfo_id())`

  - get scale-factor for windows https://stackoverflow.com/a/59628224
- [x] add missing import `from selenium.webdriver.remote.webelement import WebElement` at https://vinyzu.gitbook.io/cdp-patches-documentation/input/selenium-usage#sync-usage-selenium-sync-selenium-driverless
- [x] fix selenium test https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/blob/4cc2015b8f76591436c01ad9869e0592d248ffa6/tests/test_selenium.py#L19-L41 (or remove support?)
- [x] update relevant documentation for https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/commit/e8d7437453ad2e5904e14d8d4ddaf236f5492e4d
- [x] invite @kaliiiiiiiiii to [documentation](https://vinyzu.gitbook.io/cdp-patches-documentation) on gitbook
- [x] add custom exception for when chrome has already been closed//quit

**Comment by Vinyzu** [comment]
https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/pull/5

**Comment by Vinyzu** [comment]
https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/commit/144bd953b52e67932974ba2eb34e7269780d83a1

**Comment by Vinyzu** [comment]
https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/commit/de86681c327f5d165a4f312d4f0ceba2286c3755

**Comment by Vinyzu** [comment]
https://github.com/Kaliiiiiiiiii-Vinyzu/CDP-Patches/commit/7d99bcf1908fd9caaf218801460245bcb0af9661
